### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by n-lebedev** [javascript-obfuscator/webpack-obfuscator#43]
After trying to create a build via webpack produces an 
_**Uncaught RangeError error: Maximum call stack size exceeded**_

**File**
`console.log('hi')`

**Webpack Config**
`const path = require('path');

/**
 * Connect plugins
 */

const htmlWebpackPlugin = require('html-webpack-plugin');
const javaScriptObfuscator = require('webpack-obfuscator');

/**
 * Webpack config
 */

module.exports = {
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'build'),
        filename: 'bundle.js'
    },
    plugins: [
        new htmlWebpackPlugin({
            template: './index.html',
            inject: false
        }),
        new javaScriptObfuscator({
            compact: true,
            identifierNamesGenerator: 'mangled',
            renameGlobals: true,
            selfDefending: true,
            controlFlowFlattening: true,
            controlFlowFlatteningThreshold: 1,
            deadCodeInjection: true,
            deadCodeInjectionThreshold: 1,
            stringArray: true,
            rotateStringArray: true,
            stringArrayEncoding: 'rc4',
            stringArrayThreshold: 1,
            transformObjectKeys: true,
            unicodeEscapeSequence: true,
            target: 'browser'
        })
    ]
};`

If you turn off the Self Defending everything works

**Comment by roitto** [comment]
I have this same issue too. It appeared after updating to webpack 4.

**Comment by peterwilli** [comment]
@roitto @n-lebedev Little late but you can add

```javascript
optimization: {
	// We no not want to minimize our code.
	minimize: false
},
```

To your webpack config to disabled internal minification, this will make the script work again with self defending on.

**Comment by zamotkin** [comment]
Should be fixed in version 0.18.7
